torch_geometric/nn/conv/gat_conv.py

```diff
@@ -176,6 +181,21 @@ def __init__(
             self.lin_edge = None
             self.register_parameter('att_edge', None)
 
+        if residual and concat:
```

Comment:
Maybe we can make this cleaner. For example, we can define something like
```python
if concat:
    out_channels = out_channels * heads
if residual and in_channels != out_channels:
    self.lin_res = Linear(in_channels, out_channels, ...)
elif residual:
    self.lin_res = Identity()
else:
    self.register_parameter('lin_res', None)
```

Reply:
I think in GATConv, out_channels means out channels per head. Let out_channels = out_channels * heads will require a refactor of other usages of out_channels in GATConv.

I can help adjust this if it's necessary.

Reply:
You can use a different name

Reply:
Sure. 

Since the `out_channels` behaves the same for `residual` and `bias`, I also use the new variable `out_channels_total` to simplify the codes for `bias`. 
No other usage of `heads * out_channels` should be replaced by this new variable now.